src/main/java/fact/utils/PhotonStream2ArrivalTime.java

```diff
@@ -0,0 +1,55 @@
+package fact.utils;
+
+import java.util.Arrays;
+import stream.Data;
+import stream.Processor;
+import stream.annotations.Parameter;
+
+public class PhotonStream2ArrivalTime implements Processor {
+    @Parameter(
+        required = true,
+        description = "The arrival slices of the single pulses.")
+    private String singlePulsesKey = null;
+
+    @Parameter(
+        required = true,
+        description = "The reconstruted arrival time")
+    private String arrivalTimeKey = null;
+
+    @Override
+    public Data process(Data input) {
+        int[][] singlePulses = (int[][]) input.get(singlePulsesKey);
+        double[] arrivalTimes = new double[singlePulses.length];
+
+        for (int pix = 0; pix < singlePulses.length; pix++) {
+            arrivalTimes[pix] = this.getMedian(singlePulses[pix]);
+        }
+        input.put(arrivalTimeKey, arrivalTimes);
+        return input;
+    }
+
+    public double getMedian(int[] pulses){
```

Comment:
Ah I see .. I just Ctrl-f searched the whole fact-tools code for the word "median" and found such code in `MuonRingClean.java` so I assumed this is the way to go in fact-tools.